### ISSUE TITLE
Many errors during CI tests when pushing139b393 to github, on 'Moodle Code Checker' step

### ISSUE THREAD

**Issue by dcadiou** [grp-attestoodle/moodle-tool_attestoodle#170]
See https://github.com/grp-attestoodle/moodle-tool_attestoodle/actions/runs/11350074955/job/31567663549 (if still available).

Otherwise :
[logs_29650713151.zip](https://github.com/user-attachments/files/17381829/logs_29650713151.zip)

**Comment by dcadiou** [comment]
Thoses errors could also been seen using the Moodle plugin https://moodle.org/plugins/local_codechecker if it is installed in the Moodle used (by visiting https://..../local/codechecker/?path=admin%252Ftool%252Fattestoodle&includewarnings=0&showstandard=1)

**Comment by dcadiou** [comment]
The last commit pushed solved all the errors previously raised, but checks continue to fail because of warnings.

I was suspecting that warnings doesn't generate a failure during push to github, but I was wrong.

To continue on this issue, I now have to fix also all the more than 400 warnings that still exists, unfortunatly.

**Comment by dcadiou** [comment]
# Fix using CodeSniffer itself

I discoved that CodeSniffer, used by Moodle.org for theses checks ('Moodle Code Checker'), can also be used to correct problems.

Requirement = Moodle plugin https://github.com/moodlehq/moodle-local_codechecker must be installed in the Moodle used.

Then checks can be reruned with 

```sh
cd /path/to/moodle/root/folder
php local/codechecker/vendor/squizlabs/php_codesniffer/bin/phpcs --config-set default_standard moodle
php local/codechecker/vendor/squizlabs/php_codesniffer/bin/phpcs admin/tool/attestoodle
```

and corrections be done with

```sh
php local/codechecker/vendor/squizlabs/php_codesniffer/bin/phpcbf -v -p admin/tool/attestoodle/
```
See the following execution log for details : 
[phpcbf_2024-10-17_11h59.log](https://github.com/user-attachments/files/17410144/phpcbf_2024-10-17_11h59.log)

**Comment by dcadiou** [comment]
Code sniffer was not able to automatically fix all warnings. Some remains and has to be fixed manually.

**Comment by dcadiou** [comment]
Merged into master via pull request #172
